### ISSUE TITLE
abcjs basic obscures obscures lower staves in Safari

### ISSUE THREAD

**Issue by dvdrndlph** [paulrosen/abcjs#35]
I am making a call like this:
```
ABCJS_RENDERER_SETTINGS = {scale:0.6, paddingtop:10, paddingbottom:1};
ABCJS.renderAbc('notation0', abc_str, {}, ABCJS_RENDERER_SETTINGS);
```
to render an abc_str like this:
```
X:1
L:1/16
M:2/4
V:1 treble  
V:2 bass
%%annotationfont   Helvetica  6
K:C
V:1
z CDE {def}[F D E C]2 [fdec][fdc] | G2c2 B2c2 | dGAB cABG 
V:2
z8 | z C,D,E, F,D,E,C, | G,2G,,2 A,2B,2
```
This works fine in the Chrome browser. But most of the bass clef is cut off when I try it in Safari. It seems that abcjs is setting some incorrect style properties--making the height shorter than the SVG to be rendered and then asking the overflow to be hidden. I see this in the generated HTML:

```
<div id="notation0" style="width: 755px; overflow: hidden; height: 120.305068852353px;">
<svg height="200.5084480872555" version="1.1" width="1258.3333333333335"
xmlns="http://www.w3.org/2000/svg" style="overflow: hidden; 
....
```

It seems that Chrome is ignoring the style settings, but Safari is enforcing them. This is a bit of a show-stopper for me.

**Comment by dvdrndlph** [comment]
This seems to be a scaling problem. When scale is 1.0, the problem (horizontal) problem goes away. But then the score is clipped off on the right. As the scale decreases, more and more of the bottom of the score is cut off.

**Comment by paulrosen** [comment]
Ok, I'll look at it soon.

In the meantime, you can work around this by setting the height yourself
right after the call to ABCJS.


On Tue, May 12, 2015 at 1:00 PM, David Randolph <notifications@github.com>
wrote:

> This seems to be a scaling problem. When scale is 1.0, the problem
> (horizontal) problem goes away. But then the score is clipped off on the
> right. As the scale decreases, more and more of the bottom of the score is
> cut off.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/paulrosen/abcjs/issues/35#issuecomment-101348342>.
>



-- 
Paul Rosen
Performant Software Solutions, LLC

**Comment by dvdrndlph** [comment]
Another thing I see is a big jump in the size of a score with scale:0.82 compared to one with scale:0.81.

**Comment by dvdrndlph** [comment]
The following workaround is working fine for me now:
```
var div = document.getElementById(renderer_name);
    div.style.height = '250px';
    div.style.height = '950px';
    div.style.overflow = 'auto'; // just in case anything goes missing
```
I see that the ``%%annotation font`` and ``%%sysstaffsep`` directives are working now. That is really good news.

Thanks,
Dave

**Comment by paulrosen** [comment]
Heh, it was just a typo. It will be fixed in version 2.2, or you can download the "master" branch now.

I didn't see the jump you mentioned between .81 and .82, but perhaps that was an artifact of the other bug. Let me know if you still see it.

Yes, the last release contained a number of new formatting directives.
